### DISCUSSION TITLE
Git Fetch command hangs while processing vcpkg install

### DISCUSSION THREAD

**Discussion by drathin4** [microsoft/vcpkg#27528] [Q&A] [answered]
I'm in the process of integrating a custom library in `VCPkg` and I'm using `vcpkg-registry` for that. I've already established communication with the support team previously in that regard:

[https://github.com/microsoft/vcpkg/discussions/27308](url)

I'm facing a peculiar problem currently. I closed all the windows and restarted my machine after system updates and post that event this particular problem occurred. I used the command `vcpkg install --feature-flags=registries,manifests`, but then the `fetch` command just hangs. Below is the CMD output:

Detecting compiler hash for triplet x64-windows...
The following packages will be built and installed:
    fleck[core]:x64-windows -> 1.0 -- C:\Users\SEZ5505\AppData\Local\vcpkg\registries\git-trees\85e5a9a4ff38e5312e2b4bdc692871de836579cd
Restored 0 package(s) from C:\Users\SEZ5505\AppData\Local\vcpkg\archives in 112.2 us. Use --debug to see more details.
Installing 1/1 fleck:x64-windows...
Building fleck[core]:x64-windows...
-- Installing port from location: C:\Users\SEZ5505\AppData\Local\vcpkg\registries\git-trees\85e5a9a4ff38e5312e2b4bdc692871de836579cd
-- Fetching https://dev.azure.com/neilsoft-innovation/scan2bim/_git/finitor 47d39f860aef08820946ff300f610c3c77427f52...

Occasionally, I also get the following error:

fatal: Unable to create 'D:/dev/vcpkg/downloads/git-tmp/.git/shallow.lock': File exists.

Another git process seems to be running in this repository, e.g. an editor opened by 'git commit'. Please make sure all processes are terminated then try again. If it still fails, a git process may have crashed in this repository earlier: remove the file manually to continue.

I removed the `shallow.lock` file, but again the CMD hangs while fetching the library from Azure. I also tried following the steps on the below website (including removal of `.git\index` file), but to no avail:

[https://www.positioniseverything.net/another-git-process-seems-to-be-running-in-this-repository/](url)

I'm hoping for inputs from your end. Thanks.

**Comment by BillyONeal** [comment, marked as answer]
Unfortunately, because we do not see this problem elsewhere, you need to do some digging to figure out what else on your system is trying to touch this repo at the same time. Something like procmon may be able to help you: https://learn.microsoft.com/en-us/sysinternals/downloads/procmon

**Reply by drathin4** [reply]
Well, all of a sudden after the weekend, this issue has resolved without any changes from my end ! If it happens again, I will consider installing procmon as you suggested. Thanks.
